Add CLI for creating new component.

jdpedrie opened the pull request:
To simplify the process of adding a new component and to make it easier for those not as familiar with the workings of Google Cloud PHP, we're adding an interactive CLI to configure new components.

This CLI is built with the intention of being used for gapic-only clients, but will work with handwritten veneers as well. I've tried to include explanations and suggestions wherever possible to make it as straightforward as possible.

Running it is simple:
```
$ dev/google-cloud component
```

I've created a branch with Bigtable completely uninitialized -- that is, the gapics are present, but no component setup has been done, in order to allow you to try it out. You can find it here: https://github.com/jdpedrie/google-cloud-php/tree/bigtable-uninitialized

michaelbausor commented:
This is GREAT! Some feedback:
  - Docs url should trim trailing slash (because i entered the cloud site url with a trailing slash, the suggested docs url had two slashes in it)
  - Feature request: accept short name as first input (e.g. bigtable) and then suggest defaults for component name, display name, cloud uri, etc.
  - Yes/No questions should indicate whether Yes or No is the default if you just hit enter. Maybe when suggestion is NOT to add, default No, otherwise default Yes?
  - Default git repo should be GoogleCloudPlatform/google-cloud- **php-** bigtable.git
  - There were a lot of options that I think you would always be able to accept the default if you are doing a gapic-only API. Maybe there could be an early question like "Would you like to accept all remaining GAPIC-only defaults?", and then fast-forward. The only options that we would need to be careful to exclude from such a fast-forward process would be those that might vary based on an unusual name (like needing to know that something should be called "Error Reporting" instead of "Errorreporting"). But for a GAPIC-only with a simple 1-word name, a fast-forward option would be nice.

To reiterate though, this is GREAT!! :-)

michaelbausor commented:
Also, I don't think I got a CONTRIBUTING file?

jdpedrie commented:
Thanks for the feedback @michaelbausor. I'll work on those changes tomorrow!

jdpedrie commented:
@michaelbausor 

> Docs url should trim trailing slash (because i entered the cloud site url with a trailing slash, the suggested docs url had two slashes in it)

Fixed this!

> Feature request: accept short name as first input (e.g. bigtable) and then suggest defaults for component name, display name, cloud uri, etc.

Great idea! Implemented and it definitely helps speed up the process.

> Yes/No questions should indicate whether Yes or No is the default if you just hit enter. Maybe when suggestion is NOT to add, default No, otherwise default Yes?

Fixed this.

> Default git repo should be GoogleCloudPlatform/google-cloud- **php-** bigtable.git

:+1:

> There were a lot of options that I think you would always be able to accept the default if you are doing a gapic-only API. Maybe there could be an early question like "Would you like to accept all remaining GAPIC-only defaults?", and then fast-forward. The only options that we would need to be careful to exclude from such a fast-forward process would be those that might vary based on an unusual name (like needing to know that something should be called "Error Reporting" instead of "Errorreporting"). But for a GAPIC-only with a simple 1-word name, a fast-forward option would be nice.

This one is a little bit tougher. I'll work on it some more. In the meantime though, I did some work on improving the general intelligence of the CLI, providing better validation and defaults, and skipping a number of prompts that were unneccessary, especially in the readme config section.

jdpedrie commented:
Also, I updated the bigtable-uninitialized branch on my fork to the latest version of the CLI, so you should be able to try out the new changes.

michaelbausor commented:
Just tried it again. Worked really well! One small issue:
  - Somehow I ended up with a composer component path field that was an absolute path, starting from the root on my machine, instead of just `src/Bigtable`. I'm not sure how I did that exactly though.
  - Re the simple GAPIC-only option that may be complex to implement - I think just leave it as-is for now. It is more that good enough, especially as I expect to be the one using it most of the time. We can prioritize a more streamlined process in future if there is a need.

michaelbausor commented:
LGTM. We may want to revisit as we refine our process in future, but as it stands this is extremely useful, so I think merging into master makes sense.
